public/app/features/transformers/partitionByValues/partitionByValues.test.ts

```diff
@@ -184,4 +184,73 @@ describe('Partition by values transformer', () => {
     expect(partitioned[2].name).toEqual('XYZ region=China status=OK');
     expect(partitioned[3].name).toEqual('XYZ region=China status=FAIL');
   });
+
+  it('should partition by multiple fields naming: {asLabels: true}', () => {
+    const source = [
+      toDataFrame({
+        name: 'XYZ',
+        refId: 'A',
+        fields: [
+          { name: 'model', type: FieldType.string, values: ['E1', 'E2', 'C1', 'E3', 'C2', 'C3'] },
+          { name: 'region', type: FieldType.string, values: ['Europe', 'Europe', 'China', 'Europe', 'China', 'China'] },
+          { name: 'status', type: FieldType.string, values: ['OK', 'FAIL', 'OK', 'FAIL', 'OK', 'FAIL'] },
+        ],
+      }),
+    ];
+
+    const config: PartitionByValuesTransformerOptions = {
+      fields: ['region', 'status'],
+      naming: {
+        asLabels: true,
+      },
+    };
+
+    let partitioned = partitionByValuesTransformer.transformer(config, ctx)(source);
+
+    // all frame names are same
+    expect(partitioned[0].name).toEqual('XYZ');
+    expect(partitioned[1].name).toEqual('XYZ');
+    expect(partitioned[2].name).toEqual('XYZ');
+    expect(partitioned[3].name).toEqual('XYZ');
+
+    // all frames contain all fields
+    expect(partitioned[0].fields[0].name).toEqual('model');
+    expect(partitioned[0].fields[1].name).toEqual('region');
+    expect(partitioned[0].fields[2].name).toEqual('status');
+
+    // in each frame, every field has same labels
+    expect(partitioned[0].fields[0].labels).toEqual({ region: 'Europe', status: 'OK' });
+    expect(partitioned[1].fields[0].labels).toEqual({ region: 'Europe', status: 'FAIL' });
+    expect(partitioned[2].fields[0].labels).toEqual({ region: 'China', status: 'OK' });
+    expect(partitioned[3].fields[0].labels).toEqual({ region: 'China', status: 'FAIL' });
+  });
+
+  it('should partition by multiple fields and omit those fields in result', () => {
+    const source = [
+      toDataFrame({
+        name: 'XYZ',
+        refId: 'A',
+        fields: [
+          { name: 'model', type: FieldType.string, values: ['E1', 'E2', 'C1', 'E3', 'C2', 'C3'] },
+          { name: 'region', type: FieldType.string, values: ['Europe', 'Europe', 'China', 'Europe', 'China', 'China'] },
+          { name: 'status', type: FieldType.string, values: ['OK', 'FAIL', 'OK', 'FAIL', 'OK', 'FAIL'] },
+        ],
+      }),
+    ];
+
+    const config: PartitionByValuesTransformerOptions = {
+      fields: ['region', 'status'],
+      naming: {
+        asLabels: true,
+      },
+      omitFields: true,
+    };
+
+    let partitioned = partitionByValuesTransformer.transformer(config, ctx)(source);
+
+    // all frames contain only model field
+    expect(partitioned[0].fields.length).toEqual(1);
+    expect(partitioned[0].fields[0].name).toEqual('model');
+    expect(partitioned[0].fields[0].labels).toEqual({ region: 'Europe', status: 'OK' });
+  });
 });
```

Comment:
We talked about in daily today that it may be valuable to include a gdev dashboard for this transform so we can more easily share this change / educate users on value / usage of transforms in general

Reply:
yea, that great!!

Reply:
We have some transform test dashboards (and panel on the overview dashboard that filters on them) 
<img width="377" alt="Screenshot 2023-01-14 at 09 23 26" src="https://user-images.githubusercontent.com/10999/212463259-effbd1c4-a4c7-432d-b5bd-54cbb963afe4.png">

Reply:
we discussed transform dashboards today with @bohandley and @gtk-grafana and have some bigger plans for visualizing transforms (something similar to Debug Snapshot dashboard preview).

i'm gonna hold off adding one in this PR for now.